parametric_stellarator.py

```diff
@@ -373,43 +375,106 @@ def offset_surface(vmec, theta, phi, offset, period_ext):
 
         # Compute offset surface point
         r = offset_point(vmec, zeta, theta, offset)
+    
+    # Raise error for negative offset values
+    elif offset < 0:
+        raise ValueError(
+            'Offset must be greater than or equal to 0. Check thickness inputs '
+            'for negative values'
+        )
 
     return r
 
 
+def expand_offset(offset_vals, ang_list, ang_list_exp, num_ang_list):
```

Comment:
Perhaps we should use the existing SciPy capability for this: https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.RegularGridInterpolator.html#scipy-interpolate-regulargridinterpolator